mpr-research-data.py

```diff
@@ -7,65 +7,34 @@
 import sqlalchemy as sql
 from google.cloud import storage
 
-logging.basicConfig(
-    format="%(asctime)s %(levelname)-8s [%(filename)s:%(lineno)d] - %(message)s",
-    datefmt="%Y-%m-%dT%H:%M:%S%z",
-    level=logging.INFO
-)
-
-# SETUP CONFIG VARIABLES
-# --------------------------------------------------------------------------
-targetBucketName = os.getenv('GCLOUD_BUCKET', 'mpr-research-data-uploads')
-
-numberOfMonths = os.getenv('NUMBER_OF_MONTHS', 4)
-
-defaultQueryFolder = os.getenv('QUERY_FOLDER', 'queries')
-queryTemplateDict = {'course': os.getenv('COURSE_QEURY', 'courseQuery.sql'),
-                     'retrieve': os.getenv('RETRIEVE_QUERY', 'retrieveQuery.sql')}
-
 
 # FUNCTIONS
 # --------------------------------------------------------------------------
 
-def getDBCreds():
-
-    dbCredsDefaultDict = {'NAME': None,
-                          'USER': None,
-                          'PASSWORD': None,
-                          'HOST': None,
-                          'PORT': 3306}
-    dbCredsDict = {}
+def setUpLogger():
 
-    allKeyPartsFound = True
-    for credPart in dbCredsDefaultDict:
-        dbCredsDict[credPart] = os.getenv(
-            'DB_' + credPart, dbCredsDefaultDict[credPart])
-
-        if not dbCredsDict[credPart]:
-            logging.error(
-                f'Did not find .env variable for M-Write Peer Review production DB key: {credPart}')
-            allKeyPartsFound = False
-
-    if not allKeyPartsFound:
-        sys.exit('Exiting.')
-
-    return dbCredsDict
-
-
-def getGCPCreds() -> dict:
-    gcpCredsDict = json.loads(os.getenv('GCP_KEY'))
-    return gcpCredsDict
+    logWarning = False
+    raiseLogTypeError = None
 
+    try:
+        logLevel = str(os.environ.get('LOG_LEVEL', 'DEBUG')).upper()
+    except TypeError as e:
+        logWarning = True
+        raiseLogTypeError = e
+    
+    if logLevel not in ['DEBUG', 'INFO', 'WARNING', 'ERROR', 'FATAL', 'CRITCAL', 'EXCEPTION']:
+        logWarning = True
```

Comment:
A couple of thoughts about this log level validation…

1. As I mentioned in comments for #15, maybe don't validate it.  Try using it and fall back to a default level (`INFO`) if an exception is thrown about an invalid value.
2. If you think this validation is important and should be done here, there's a good programmatic way to get the valid level names:
   ```suggestion
       if logLevel not in [logging.getLevelName(10 * i) for i in range(1, 6)]:
           logWarning = True
   ```

   That gives the five basic log levels:

   ```pycon
   >>> [logging.getLevelName(10 * l) for l in range(1, 6)]
   ['DEBUG', 'INFO', 'WARNING', 'ERROR', 'CRITICAL']
   ```
   
   I think `EXCEPTION` isn't a log level.  If it's something new, please point out the documentation to me.  The level `FATAL` is an alias for `CRITICAL`, so it's unnecessary.

Reply:
I'm just going to remove validation to simplify things, I like the idea to keep it pythonic.

Reply:
✅